doc/Language/js-nutshell.pod6

```diff
@@ -0,0 +1,191 @@
+say 'Hello, world!';
+=end code
+
+Parentheses are optional for function calls in Perl 6. While semicolons are,
+for the most part, optional in Node.js, they are mandatory for expressions in
+Perl 6.
+
+Now that we've greeted the world, let's greet our good friend, Joe. We'll
+start with Node.js again:
+
+=begin code :lang<javascript>
+let name = 'Joe';
+console.log('What\'s up,' + name + '?');
+console.log(`What's up, {name}?`);
+console.log("What's up, ", name, "?");
+=end code
+
+Since he didn't hear us, let's greet him again, this time in Perl 6:
+
+=begin code
+my $name = 'Joe';
+say 'What\'s up, ' ~ $name ~ '?';
+say "What's up, $name?";
+say "What's up, ", $name, "?";
+=end code
+
+Here, there are only a couple differences: most variables in Perl 6 have what
+are called sigils, which are what the C<$> in front of its name is, and string
+concatenation uses the C<~> operator instead of C<+>. What the two languages
+share in common here is support for string interpolation.
+
+Now that the basic examples are out of the way, let's explain the similarities
+between the two languages in greater detail.
+
+=head2 Variables
+
+Variables in Node.js can be defined like this;
+
+=begin code :lang<javascript>
+var foo = 1;    // Lexically scoped with functions and modules
+let foo = 1;    // Lexically scoped with blocks
+const foo = 1;  // Lexically scoped with blocks; constant
+
+global.foo = 1; // Dynamically scoped; global
+foo = 1;        // Ditto, but implicit; forbidden in strict mode
+=end code
+
+In Perl 6 there is no equivalent to C<var>. An important note to make is that
+there is no variable hoisting in Perl 6; variables are defined and assigned
+at the line they're on, not defined at the top of its scope and later assigned
+at that line.
+
+This is how the equivalent types of variables are defined in Perl 6:
+
+=begin code
+my $foo = 1;      # Lexically scoped with blocks
+our $foo = 1;     # Lexically scoped with blocks and modules
+constant foo = 1; # Lexically scoped with blocks and modules; constant
+
+my $*foo = 1;       # Dynamically scoped with blocks
+OUR::<$foo> = 1;    # Dynamically scoped with blocks and modules
+GLOBAL::<$foo> = 1; # Dynamically scoped; global
+=end code
+
+Use C<my> where you'd use C<let>, C<our> for variables you'd define in the
+outermost scope needed, and C<constant> where you'd uses C<const>.
+
+Dynamically scoped variables are not referred to in the same way as lexically
+scoped ones like they are in Node.js. User-defined ones use either a C<$*>,
+C<@*>, C<%*>, or C<&*> twigil. Refer to the documentation on
+L<variables|/language/variables> for more information on sigils, twigils, and
+variable containers.
+
+Variables in Node.js can override others from outer scopes with the same name
+(though linters will usually complain about it depending on how they're
+configured):
+
+=begin code :lang<javascript>
+let foo = 1;
+function logDupe() {
+    let foo = 2;
+    console.log(foo);
+}
+logDupe(2); // 2
```

Comment:
Maybe add `console.log(foo)` here to show that the original `foo` declared before the `function` is unaffected.